src/r3d.imageio/r3dinput.cpp

```diff
@@ -0,0 +1,441 @@
+// Copyright Contributors to the OpenImageIO project.
+// SPDX-License-Identifier: Apache-2.0
+// https://github.com/AcademySoftwareFoundation/OpenImageIO
+
+// R3D SDK can be downloaded from the following site:
+// https://www.red.com/download/r3d-sdk
+//
+// The code has been tested with the version 8.5.1 installed in
+// /opt/R3DSDKv8_5_1 directory and setting up the variable
+// export R3DSDK_ROOT="/opt/R3DSDKv8_5_1"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdio>
+
+#include <OpenImageIO/filesystem.h>
+#include <OpenImageIO/fmath.h>
+#include <OpenImageIO/imageio.h>
+#include <OpenImageIO/tiffutils.h>
+
+#include <R3DSDK.h>
+#include <R3DSDKCuda.h>
+#include <R3DSDKDefinitions.h>
+
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include <cuda_runtime.h>
+#include <vector>
+
+#include <condition_variable>
+#include <functional>
+#include <iostream>
+#include <list>
+#include <map>
+#include <mutex>
+#include <thread>
+#include <time.h>
+#include <unistd.h>
+#include <vector>
+
+unsigned char*
+AlignedMalloc(size_t& sizeNeeded)
+{
+    // alloc 15 bytes more to make sure we can align the buffer in case it isn't
+    unsigned char* buffer = (unsigned char*)malloc(sizeNeeded + 15U);
+
+    if (!buffer)
+        return nullptr;
+
+    sizeNeeded = 0U;
+
+    // cast to a 32-bit or 64-bit (depending on platform) integer so we can do the math
+    uintptr_t ptr = (uintptr_t)buffer;
+
+    // check if it's already aligned, if it is we're done
+    if ((ptr % 16U) == 0U)
+        return buffer;
+
+    // calculate how many bytes we need
+    sizeNeeded = 16U - (ptr % 16U);
+
+    return buffer + sizeNeeded;
+}
+
+OIIO_PLUGIN_NAMESPACE_BEGIN
+
+#define DBG if (1)
```

Comment:
I think we want to turn debug messages off before merging this.

Reply:
ACK

Reply:
C001

Reply:
Will try it with the environment variable controling the debug level verbosity.